src/vs/platform/externalTerminal/node/externalTerminalService.ts

```diff
@@ -213,8 +213,14 @@ export class LinuxExternalTerminalService extends ExternalTerminalService implem
 	}
 
 	public runInTerminal(title: string, dir: string, args: string[], envVars: ITerminalEnvironment, settings: IExternalTerminalSettings): Promise<number | undefined> {
-
-		const execPromise = settings.linuxExec ? Promise.resolve(settings.linuxExec) : LinuxExternalTerminalService.getDefaultTerminalLinuxReady();
+		let execPromise: Promise<string>;
+		if (process.env.WSL_DISTRO_NAME) {
```

Comment:
Debugging seemed to workwhen I set linuxExec to `"wt"`, this should be equivalent to that.

Reply:
Yeah but I wonder whether it starts in the windows context (this is what happens when I run wt.exe in my WSL), then runs `bash` (line 236) which will run the command in WSL, but if you have multiple distros, it might have actually started the wrong one?

Reply:
Yeah that makes sense, will change